plugin-jdbc-duckdb/src/main/java/io/kestra/plugin/jdbc/duckdb/Query.java

```diff
@@ -91,18 +99,30 @@ public void registerDriver() throws SQLException {
     }
 
     @Override
+    @Schema(
+        title = "The JDBC URL to connect to the database",
+        description = "The default value, `jdbc:duckdb:`, will use a local in-memory database. \nSet this property when connecting to a persisted database instance, for example `jdbc:duckdb:md:my_database?motherduck_token=<my_token>` to connect to [MotherDuck](https://motherduck.com/).",
+        defaultValue = DEFAULT_URL
+    )
+    @NotNull
     public String getUrl() {
-        return "jdbc:duckdb:" + databaseFile;
+        if (DEFAULT_URL.equals(this.url)) {
+            return DEFAULT_URL + this.databaseFile;
+        }
+        return this.url;
     }
 
     @Override
     public Query.Output run(RunContext runContext) throws Exception {
         Map<String, String> outputFiles = null;
 
-        this.databaseFile = runContext.tempFile();
-        Files.delete(this.databaseFile);
+        // we only create the database file if the default URL is used.
+        if (DEFAULT_URL.equals(this.url)) {
+            this.databaseFile = runContext.tempFile();
+            Files.delete(this.databaseFile);
```

Comment:
Why do we create a tempFile to delete it right afterwards (I know it was already there before but trying to understand)

Reply:
No idea, I think it's here to be sure that the file didn't exist as DuckDB will create it.

Reply:
duckdb expect to have an empty file for it's database, so this ugly method is only to have a temporary file path